pallets/rmrk-core/src/functions.rs

```diff
@@ -405,19 +418,20 @@ where
 	fn nft_mint_directly_to_nft(
 		sender: T::AccountId,
 		owner: (CollectionId, NftId),
+		nft_id: NftId,
 		collection_id: CollectionId,
 		royalty_recipient: Option<T::AccountId>,
 		royalty_amount: Option<Permill>,
 		metadata: StringLimitOf<T>,
 		transferable: bool,
-		resources: Option<BoundedResourceTypeOf<T>>,
+		resources: Option<BoundedResourceInfoTypeOf<T>>,
 	) -> sp_std::result::Result<(CollectionId, NftId), DispatchError> {
-		let nft_id = Self::get_next_nft_id(collection_id)?;
+		ensure!(!Self::nft_exists((collection_id, nft_id)), Error::<T>::NftAlreadyExists);
 		let collection = Self::collections(collection_id).ok_or(Error::<T>::CollectionUnknown)?;
 
-		// Prevent minting when next NFT id is greater than the collection max.
+		// Prevent minting when nfts_count is greater than the collection max.
 		if let Some(max) = collection.max {
-			ensure!(nft_id < max, Error::<T>::CollectionFullOrLocked);
+			ensure!(collection.nfts_count < max, Error::<T>::CollectionFullOrLocked);
```

Comment:
Should we add a helper function that gets the Collection's NFT count?

Reply:
```
pub fn get_collection_nfts_count(collection_info: CollectionInfoOf<T>) -> u32 {
		collection_info.nfts_count
}
```

Reply:
I am not sure actually, seems a bit unneeded abstraction of just accessing property? But happy to update if you think worth pursuing. ( Usually I would extract a function with more logic )